voluptuous/humanize.py

```diff
@@ -0,0 +1,39 @@
+from voluptuous import Invalid, MultipleInvalid
+from voluptuous.error import Error
+
+
+MAX_VALIDATION_ERROR_ITEM_LENGTH = 500
+
+
+def _nested_getitem(data, path):
+    for item_name in path:
+        try:
```

Comment:
here's what I ended up with; it's kind of the posterchild of why EAFP is sometimes more straightforward and thus safer in Python:
```python
def _nested_getitem(data, path):
    for item_index in path:
        # Data could be a dict or indexable sequence (list/tuple)
        item_present = (
            isinstance(data, dict) and item_index in data
            or item_index > (len(data) - 1)
        )
        if item_present:
            data = data[item_index]
        else:
            return None
    return data
```

I can go with that if you prefer (it does have the advantage of being more explicit about lists - maybe I should add a test case for that anyway) but I think the `try: except` is more straightforward because you don't have to reverse-engineer what's going to actually happen when you try to index.